code/modules/mod/mod.dm

```diff
@@ -0,0 +1,541 @@
+	/// MOD chestplate.
+	var/obj/item/clothing/suit/armor/mod/chestplate
+	/// MOD gauntlets.
+	var/obj/item/clothing/gloves/mod/gauntlets
+	/// MOD boots.
+	var/obj/item/clothing/shoes/mod/boots
+	/// List of parts.
+	var/list/mod_parts = list()
+	/// Modules the MOD should spawn with.
+	var/list/initial_modules = list()
+	/// Modules the MOD currently possesses.
+	var/list/modules = list()
+	/// Currently used module.
+	var/obj/item/mod/module/selected_module
+	/// AI mob inhabiting the MOD.
+	var/mob/living/silicon/ai/ai
+	/// Delay between moves as AI.
+	var/movedelay = 0
+	/// Cooldown for AI moves.
+	COOLDOWN_DECLARE(cooldown_mod_move)
+	/// Person wearing the MODsuit.
+	var/mob/living/carbon/human/wearer
+
+/obj/item/mod/control/Initialize(mapload, newtheme)
+	. = ..()
+	if(newtheme)
+		theme = newtheme
+	theme = GLOB.mod_themes[theme]
+	slowdown = theme.slowdown_unactive
+	complexity_max = theme.complexity_max
+	skin = theme.default_skin
+	cell_drain = theme.cell_usage
+	wires = new /datum/wires/mod(src)
+	if(req_access?.len)
+		locked = TRUE
+	if(ispath(cell))
+		cell = new cell(src)
+	if(ispath(theme.helmet_path))
+		helmet = new theme.helmet_path(src)
+		helmet.mod = src
+		mod_parts += helmet
+	if(ispath(theme.chestplate_path))
+		chestplate = new theme.chestplate_path(src)
+		chestplate.mod = src
+		mod_parts += chestplate
+	if(ispath(theme.gauntlets_path))
+		gauntlets = new theme.gauntlets_path(src)
+		gauntlets.mod = src
+		mod_parts += gauntlets
+	if(ispath(theme.boots_path))
+		boots = new theme.boots_path(src)
+		boots.mod = src
+		mod_parts += boots
+	var/list/all_parts = mod_parts.Copy() + src
+	for(var/obj/item/piece as anything in all_parts)
+		piece.name = "[theme.name] [piece.name]"
+		piece.desc = "[piece.desc] [theme.desc]"
+		piece.armor = getArmor(arglist(theme.armor))
+		piece.resistance_flags = theme.resistance_flags
+		piece.max_heat_protection_temperature = theme.max_heat_protection_temperature
+		piece.min_cold_protection_temperature = theme.min_cold_protection_temperature
+		piece.gas_transfer_coefficient = theme.gas_transfer_coefficient
+		piece.permeability_coefficient = theme.permeability_coefficient
+		piece.siemens_coefficient = theme.siemens_coefficient
+		piece.icon_state = "[skin]-[initial(piece.icon_state)]"
+	if(initial_modules.len)
+		for(var/obj/item/mod/module/module in initial_modules)
+			module = new module(src)
+			install(module, TRUE)
+	RegisterSignal(src, COMSIG_ATOM_EXITED, .proc/on_exit)
+	movedelay = CONFIG_GET(number/movedelay/run_delay)
+
+/obj/item/mod/control/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	QDEL_NULL(wires)
+	QDEL_NULL(cell)
+	if(helmet)
+		helmet.mod = null
+		QDEL_NULL(helmet)
+	if(chestplate)
+		chestplate.mod = null
+		QDEL_NULL(chestplate)
+	if(gauntlets)
+		gauntlets.mod = null
+		QDEL_NULL(gauntlets)
+	if(boots)
+		boots.mod = null
+		QDEL_NULL(boots)
+	for(var/obj/item/mod/module/module in modules)
+		module.mod = null
+		QDEL_NULL(module)
```

Comment:
If you're deleting the module right afterwards, then do you need to set `module.mod` to null?